tietokanta/src/main/resources/db/migration/V1_850__.sql

```diff
@@ -0,0 +1,87 @@
+-- toteutunut_tyo tauluun siirretään kustannuksia, yksinkertaistamaan toteutuneiden kustannusten seurantaa
+
+CREATE TABLE toteutunut_tyo
+(
+    id                  serial primary key, -- sisäinen ID
+    vuosi               smallint not null CHECK (1900 < vuosi AND vuosi < 2200),
+    kuukausi            smallint not null CHECK (13 > kuukausi AND kuukausi > 0),
+    summa               numeric,
+    tyyppi              toteumatyyppi,
+    tehtava             integer REFERENCES toimenpidekoodi (id),
+    tehtavaryhma        integer REFERENCES tehtavaryhma (id),
+    toimenpideinstanssi integer  not null REFERENCES toimenpideinstanssi (id),
+    sopimus_id          integer REFERENCES sopimus (id),
+    urakka_id           integer REFERENCES urakka (id),
+    luotu               timestamp,
+    muokattu            timestamp,
+    muokkaaja           integer references kayttaja (id),
+    unique (toimenpideinstanssi, tehtava, sopimus_id, urakka_id, vuosi, kuukausi)
+);
+
+create index toteutunut_tyo_urakka_index
+    on toteutunut_tyo (urakka_id, kuukausi, vuosi);
+
+
+COMMENT ON table toteutunut_tyo IS
+    E'Kustannusarvioitu_tyo taulusta siirretään kuukauden viimeisenä päivänä mikäli tehtäväryhmä: Erillishankinnat (W) tai tehtävät: 3054 ja 3055
+      tähän toteutunut_tyo tauluun, josta voidaan hakea toteutuneen työn kustannukset raportteihin ja kustannuksiin.';
+
+-- Siirrä kustannusarvoitu_tyo taulusta mikäli tehtäväryhmä: Erillishankinnat (W) tai tehtävät: 3054 ja 3055
+INSERT INTO toteutunut_tyo (vuosi, kuukausi, summa, tyyppi, tehtava, tehtavaryhma, toimenpideinstanssi, sopimus_id,
+                            urakka_id, luotu)
+SELECT k.vuosi,
+       k.kuukausi,
+       k.summa,
+       k.tyyppi,
+       k.tehtava,
+       k.tehtavaryhma,
+       k.toimenpideinstanssi,
+       k.sopimus,
+       (select s.urakka FROM sopimus s where s.id = k.sopimus) as "urakka-id",
+       NOW()
+FROM kustannusarvioitu_tyo k
+WHERE (SELECT (date_trunc('MONTH', format('%s-%s-%s', k.vuosi, k.kuukausi, 1)::DATE))) <
+      date_trunc('month', current_date)
+  AND (k.tehtavaryhma = (SELECT id FROM tehtavaryhma where nimi = 'Erillishankinta (W)')
+        OR k.tehtava = (select id from toimenpidekoodi t where t.nimi = 'Toimistotarvike- ja ICT-kulut, tiedotus, opastus, kokousten järjestäminen jne.') -- 3055,"Toimistotarvike- ja ICT-kulut, tiedotus, opastus, kokousten järjestäminen jne."
```

Comment:
Tähän subqueryyn t.nimi IN ('nimi1', 'nimi2')?

Reply:
Vaihettu IN () malliseen kyselyyn